movement/analysis/kinematics.py

```diff
@@ -171,6 +172,106 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
+        the same dimensions as the input data, except for ``space``
+        which will be removed.
+
+    """
+    return compute_norm(compute_velocity(data))
+
+
+def compute_path_length(
+    data: xr.DataArray,
+    start: float | None = None,
+    stop: float | None = None,
+    nan_policy: Literal["drop", "scale"] = "drop",
+    nan_warn_threshold: float = 0.2,
+) -> xr.DataArray:
+    """Compute the length of a path travelled between two time points.
+
+    The path length is defined as the sum of the norms (magnitudes) of the
+    displacement vectors between two time points ``start`` and ``stop``,
+    which should be provided in the time units of the data array.
+    If not specified, the minimum and maximum time coordinates of the data
+    array are used as start and stop times, respectively.
+
+    Parameters
+    ----------
+    data : xarray.DataArray
+        The input data containing position information in Cartesian
+        coordinates, with ``time`` and ``space`` among the dimensions.
+    start : float, optional
+        The time to consider as the path's starting point. If None (default),
+        the minimum time coordinate in the data is used.
+    stop : float, optional
+        The time to consider as the path's end point. If None (default),
+        the maximum time coordinate in the data is used.
+    nan_policy : Literal["drop", "scale"], optional
+        Policy to handle NaN (missing) values. Can be one of the ``"drop"``
+        or ``"scale"``. Defaults to ``"drop"``. See Notes for more details
+        on the two policies.
+    nan_warn_threshold : float, optional
+        If more than this proportion of values are missing in any point track,
+        a warning will be emitted. Defaults to 0.2 (20%).
+
+    Returns
+    -------
+    xarray.DataArray
+        An xarray DataArray containing the computed path length.
+        Will have the same dimensions as the input data, except for ``time``
+        and ``space`` which will be removed.
+
+    Notes
+    -----
+    Choosing ``nan_policy="drop"`` will drop NaN values from each point track
+    before computing path length. This equates to assuming that a track
+    follows a straight line between two valid points flanking a missing
+    segment. Missing segments at the beginning or end of the specified
+    time range are not counted. This approach tends to underestimate
+    the path length, and the error increases with the number of missing
+    values.
+
+    Choosing ``nan_policy="scale"`` will adjust the path length based on the
+    the proportion of valid segments per point track. For example, if only
+    80% of segments are present, the path length will be computed based on
+    these and the result will be divided by 0.8. This approach assumes
+    that motion dynamics are similar across present and missing time
+    segments, which may not be the case.
+
+    """
+    _validate_start_stop_times(data, start, stop)
```

Comment:
I agree we should rely on `slice` to do its thing and not add extra validation here. 
I've essentially removed start-stop time checks entirely, except for one case (see below).

xarray `slices`, make some specific choices:
- if start/stop time are out of the range, it only returns what's in the data's range (which is sensible)
- if start/stop are incorrect types (e.g. string), it does not error, and it silently returns an array with time dimension of size 0.

I find the 2nd choice weird and it messes up the downstream computation of path length (it will error out, with an error message that's not so easy to parse).

To account for this (time dim with size 0) and also for the edge case where the time dim has size 1 (only 1 frame selected), I now throw a somewhat informative error if the size of the time dim < 2 (in which case no meaningful path length can be computed).

I kept the test across time ranges, but I adapted it accordingly. It's useful to keep this test, to remind ourselves of the above behaviours.